### ISSUE TITLE
Networking does not come up on RHEL 7.4

### ISSUE THREAD

**Issue by superseb** [rancher/rancher#9547]
**Rancher versions:**
rancher/server: v1.6.6
rancher/agent: v1.2.5

**Infrastructure Stack versions:**
healthcheck: v0.3.0
ipsec: v0.1.2
network-services: v0.2.4
scheduler: v0.6.2
kubernetes (if applicable): n/a

**Docker version: (`docker version`,`docker info` preferred)**
Tested on upstream Docker 1.12.6, 17.03.2-ce and 17.06.0-ce
**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
RHEL 7.4 (AMI: https://access.redhat.com/articles/3135091)
`3.10.0-693.el7.x86_64`
**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
AWS
**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single node
**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle
**Steps to Reproduce:**
1. Add RHEL 7.4 host to Rancher

**Results:**
network-manager & cni-driver start, but the rest will output `Time-out getting IP address`

**Comment by leodotcloud** [comment]
Initial debugging: 

Packet capture on a non-working RHEL 7.4 host shows the reply packet not using `169.254.169.250` as the source IP address.
```

22:43:46.361947 IP (tos 0x0, ttl 64, id 42870, offset 0, flags [DF], proto UDP (17), length 96)
    172.17.0.1.57278 > 169.254.169.250.53: [bad udp cksum 0x0069 -> 0x52f5!] 5691+ A? rancher-metadata.cni-driver.ipsec.rancher.internal. (68)
        0x0000:  02fb 35f7 b8ae 0242 7763 8e6d 0800 4500  ..5....Bwc.m..E.
        0x0010:  0060 a776 4000 4011 930b ac11 0001 a9fe  .`.v@.@.........
        0x0020:  a9fa dfbe 0035 004c 0069 163b 0100 0001  .....5.L.i.;....
        0x0030:  0000 0000 0000 1072 616e 6368 6572 2d6d  .......rancher-m
        0x0040:  6574 6164 6174 610a 636e 692d 6472 6976  etadata.cni-driv
        0x0050:  6572 0569 7073 6563 0772 616e 6368 6572  er.ipsec.rancher
        0x0060:  0869 6e74 6572 6e61 6c00 0001 0001       .internal.....
22:43:46.361977 IP (tos 0x0, ttl 64, id 42871, offset 0, flags [DF], proto UDP (17), length 96)
    172.17.0.1.57278 > 169.254.169.250.53: [bad udp cksum 0x0069 -> 0x438e!] 9607+ AAAA? rancher-metadata.cni-driver.ipsec.rancher.internal. (68)
        0x0000:  02fb 35f7 b8ae 0242 7763 8e6d 0800 4500  ..5....Bwc.m..E.
        0x0010:  0060 a777 4000 4011 930a ac11 0001 a9fe  .`.w@.@.........
        0x0020:  a9fa dfbe 0035 004c 0069 2587 0100 0001  .....5.L.i%.....
        0x0030:  0000 0000 0000 1072 616e 6368 6572 2d6d  .......rancher-m
        0x0040:  6574 6164 6174 610a 636e 692d 6472 6976  etadata.cni-driv
        0x0050:  6572 0569 7073 6563 0772 616e 6368 6572  er.ipsec.rancher
        0x0060:  0869 6e74 6572 6e61 6c00 001c 0001       .internal.....
22:43:46.362186 IP (tos 0x0, ttl 64, id 59180, offset 0, flags [DF], proto UDP (17), length 132)
    172.17.0.2.53 > 172.17.0.1.57278: [bad udp cksum 0x58a7 -> 0xd005!] 9607 NXDomain*- q: AAAA? rancher-metadata.cni-driver.ipsec.rancher.internal. 0/1/0 ns: rancher.internal. [1s] SOA rancher.internal. rancher.internal. 1197 60 10 86400 1 (104)
        0x0000:  0242 7763 8e6d 02fb 35f7 b8ae 0800 4500  .Bwc.m..5.....E.
        0x0010:  0084 e72c 4000 4011 fb16 ac11 0002 ac11  ...,@.@.........
        0x0020:  0001 0035 dfbe 0070 58a7 2587 8503 0001  ...5...pX.%.....
        0x0030:  0000 0001 0000 1072 616e 6368 6572 2d6d  .......rancher-m
        0x0040:  6574 6164 6174 610a 636e 692d 6472 6976  etadata.cni-driv
        0x0050:  6572 0569 7073 6563 0772 616e 6368 6572  er.ipsec.rancher
        0x0060:  0869 6e74 6572 6e61 6c00 001c 0001 c02e  .internal.......
        0x0070:  0006 0001 0000 0001 0018 c02e c02e 0000  ................
        0x0080:  04ad 0000 003c 0000 000a 0001 5180 0000  .....<......Q...
        0x0090:  0001                                     ..
22:43:46.362219 IP (tos 0xc0, ttl 64, id 26202, offset 0, flags [none], proto ICMP (1), length 160)
    172.17.0.1 > 172.17.0.2: ICMP 172.17.0.1 udp port 57278 unreachable, length 140
        IP (tos 0x0, ttl 64, id 59180, offset 0, flags [DF], proto UDP (17), length 132)
    172.17.0.2.53 > 172.17.0.1.57278: [bad udp cksum 0x58a7 -> 0xd005!] 9607 NXDomain*- q: AAAA? rancher-metadata.cni-driver.ipsec.rancher.internal. 0/1/0 ns: rancher.internal. [1s] SOA rancher.internal. rancher.internal. 1197 60 10 86400 1 (104)[|icmp]
        0x0000:  02fb 35f7 b8ae 0242 7763 8e6d 0800 45c0  ..5....Bwc.m..E.
        0x0010:  00a0 665a 0000 4001 bb1d ac11 0001 ac11  ..fZ..@.........
        0x0020:  0002 0303 cd02 0000 0000 4500 0084 e72c  ..........E....,
        0x0030:  4000 4011 fb16 ac11 0002 ac11 0001 0035  @.@............5
        0x0040:  dfbe 0070 58a7 2587 8503 0001 0000 0001  ...pX.%.........
        0x0050:  0000 1072 616e 6368 6572 2d6d 6574 6164  ...rancher-metad
        0x0060:  6174 610a 636e 692d 6472 6976 6572 0569  ata.cni-driver.i
        0x0070:  7073 6563 0772 616e 6368 6572 0869 6e74  psec.rancher.int
        0x0080:  6572 6e61 6c00 001c 0001 c02e 0006 0001  ernal...........
        0x0090:  0000 0001 0018 c02e c02e 0000 04ad 0000  ................
        0x00a0:  003c 0000 000a 0001 5180 0000 0001       .<......Q.....
22:43:46.362296 IP (tos 0x0, ttl 64, id 59181, offset 0, flags [DF], proto UDP (17), length 132)
    172.17.0.2.53 > 172.17.0.1.57278: [bad udp cksum 0x58a7 -> 0xdf6b!] 5691 NXDomain*- q: A? rancher-metadata.cni-driver.ipsec.rancher.internal. 0/1/0 ns: rancher.internal. [1s] SOA rancher.internal. rancher.internal. 1198 60 10 86400 1 (104)
        0x0000:  0242 7763 8e6d 02fb 35f7 b8ae 0800 4500  .Bwc.m..5.....E.
        0x0010:  0084 e72d 4000 4011 fb15 ac11 0002 ac11  ...-@.@.........
        0x0020:  0001 0035 dfbe 0070 58a7 163b 8503 0001  ...5...pX..;....
        0x0030:  0000 0001 0000 1072 616e 6368 6572 2d6d  .......rancher-m
        0x0040:  6574 6164 6174 610a 636e 692d 6472 6976  etadata.cni-driv
        0x0050:  6572 0569 7073 6563 0772 616e 6368 6572  er.ipsec.rancher
        0x0060:  0869 6e74 6572 6e61 6c00 0001 0001 c02e  .internal.......
        0x0070:  0006 0001 0000 0001 0018 c02e c02e 0000  ................
        0x0080:  04ae 0000 003c 0000 000a 0001 5180 0000  .....<......Q...
        0x0090:  0001                                     ..
22:43:46.362305 IP (tos 0xc0, ttl 64, id 26203, offset 0, flags [none], proto ICMP (1), length 160)
    172.17.0.1 > 172.17.0.2: ICMP 172.17.0.1 udp port 57278 unreachable, length 140
        IP (tos 0x0, ttl 64, id 59181, offset 0, flags [DF], proto UDP (17), length 132)
    172.17.0.2.53 > 172.17.0.1.57278: [bad udp cksum 0x58a7 -> 0xdf6b!] 5691 NXDomain*- q: A? rancher-metadata.cni-driver.ipsec.rancher.internal. 0/1/0 ns: rancher.internal. [1s] SOA rancher.internal. rancher.internal. 1198 60 10 86400 1 (104)[|icmp]
        0x0000:  02fb 35f7 b8ae 0242 7763 8e6d 0800 45c0  ..5....Bwc.m..E.
        0x0010:  00a0 665b 0000 4001 bb1c ac11 0001 ac11  ..f[..@.........
        0x0020:  0002 0303 dc68 0000 0000 4500 0084 e72d  .....h....E....-
        0x0030:  4000 4011 fb15 ac11 0002 ac11 0001 0035  @.@............5
        0x0040:  dfbe 0070 58a7 163b 8503 0001 0000 0001  ...pX..;........
        0x0050:  0000 1072 616e 6368 6572 2d6d 6574 6164  ...rancher-metad
        0x0060:  6174 610a 636e 692d 6472 6976 6572 0569  ata.cni-driver.i
        0x0070:  7073 6563 0772 616e 6368 6572 0869 6e74  psec.rancher.int
        0x0080:  6572 6e61 6c00 0001 0001 c02e 0006 0001  ernal...........
        0x0090:  0000 0001 0018 c02e c02e 0000 04ae 0000  ................
        0x00a0:  003c 0000 000a 0001 5180 0000 0001       .<......Q.....
```

Packet capture on a CentOS 7 host:
```
22:53:00.177084 IP (tos 0x0, ttl 64, id 62596, offset 0, flags [DF], proto UDP (17), length 79)
    172.17.0.1.48424 > 169.254.169.250.53: [bad udp cksum 0x0058 -> 0x5e3c!] 26483+ AAAA? rancher-metadata.rancher.internal. (51)
        0x0000:  02fb 3516 e121 0242 8ee1 0606 0800 4500  ..5..!.B......E.
        0x0010:  004f f484 4000 4011 460e ac11 0001 a9fe  .O..@.@.F.......
        0x0020:  a9fa bd28 0035 003b 0058 6773 0100 0001  ...(.5.;.Xgs....
        0x0030:  0000 0000 0000 1072 616e 6368 6572 2d6d  .......rancher-m
        0x0040:  6574 6164 6174 6107 7261 6e63 6865 7208  etadata.rancher.
        0x0050:  696e 7465 726e 616c 0000 1c00 01         internal.....
22:53:00.177587 IP (tos 0x0, ttl 64, id 982, offset 0, flags [DF], proto UDP (17), length 79)
    169.254.169.250.53 > 172.17.0.1.48424: [bad udp cksum 0x0058 -> 0xd9bb!] 26483* q: AAAA? rancher-metadata.rancher.internal. 0/0/0 (51)
        0x0000:  0242 8ee1 0606 02fb 3516 e121 0800 4500  .B......5..!..E.
        0x0010:  004f 03d6 4000 4011 36bd a9fe a9fa ac11  .O..@.@.6.......
        0x0020:  0001 0035 bd28 003b 0058 6773 8580 0001  ...5.(.;.Xgs....
        0x0030:  0000 0000 0000 1072 616e 6368 6572 2d6d  .......rancher-m
        0x0040:  6574 6164 6174 6107 7261 6e63 6865 7208  etadata.rancher.
        0x0050:  696e 7465 726e 616c 0000 1c00 01         internal.....
```

**Comment by leodotcloud** [comment]
**Quick workaround:**

Upgrade the DNS service and change the command:
From: `rancher-dns --metadata-server=localhost --answers=/etc/rancher-dns/answers.json --recurser-timeout 2 --ttl 1`
To: `rancher-dns --listen=169.254.169.250:53 --metadata-server=localhost --answers=/etc/rancher-dns/answers.json --recurser-timeout 2 --ttl 1`

**Comment by imorti** [comment]
We see this on Rancher 1.6.4 as well.

**Comment by leodotcloud** [comment]
Has been merged into branch: `rancher:network-services-v025`. 
Available in network-services v0.2.5

**Comment by gui81** [comment]
I have this issue using kubernetes.  @leodotcloud mentioned a quick workaround, but how do I do this.  Is this something I can do through the rancher web interface?  If so, what are the steps?

**Comment by gui81** [comment]
I just went ahead and updated to rancher/server:v1.6.7-rc1 and everything works.  Looking forward to the stable version that fixes this.  Until then, it would be nice to somehow note that versions prior to v.1.6.7-rc1 are not compatible with RHEL 7.4 (or whatever the specific pkg update is that causes the issue).

**Comment by moelsayed** [comment]
Validation tests ran successfully using:
* rhel 7.4
* rancher v1.6.8-rc1
* docker 17.06

**Comment by moelsayed** [comment]
Validation tests passed for v1.6.8-rc1 with upstream docker 17.06 passed on RHEL 7.4

**Comment by 1026524749** [comment]
We see this on Rancher 1.6.3 as well
centos7.2 to centos7.4

**Comment by superseb** [comment]
Support for RHEL/CentOS 7.4 is available since v1.6.8.

**Comment by sagarmah** [comment]
I recently upgraded my Rancher Server - Single Container (non-HA) - NO External database from v1.5.5 to v1.6.8 to resolve some Kernel compatibility with my RHEL/CentOS 7.4. The issue was with IPSec container giving Timeout Getting IP Address Error.
I did this as per the above discussion.

I used the volume from the old container to run the new one, everything went fine, but I am still getting the same issue.
Interestingly when I just ran a fresh rancher with v1.6.8, the agents were very stable. 

Observation: Template version of IPSec stack is still old, and is not showing options to upgrade.
Can someone please help
